cpp/src/arrow/stl.h

```diff
@@ -440,7 +444,127 @@ Status TupleRangeFromTable(const Table& table, const compute::CastOptions& cast_
   return Status::OK();
 }
 
+/// \brief A STL allocator delegating allocations to a Arrow MemoryPool
+template <class T>
+class stl_allocator {
```

Comment:
Since this is in `namespace stl` now, the prefix `stl_` seems redundant. Please rename to:
```suggestion
class allocator {
```
or
```suggestion
class memory_pool_allocator {
```